consensus/babe/types.go

```diff
@@ -42,3 +42,12 @@ type AuthorityData struct {
 	id     *sr25519.PublicKey
 	weight uint64
 }
+
+// BabeHeader as defined in Polkadot RE Spec, definition 4.10 in section 4.1.4
+type BabeHeader struct {
+	Slot          uint64
+	VRFOutput     [32]byte
+	VRFProof      [32]byte
+	BlockProducer *sr25519.PublicKey
```

Comment:
the spec says this is the validator index, not the public key. practically I'm not sure it matters so much, will think about it

Reply:
actually, looks like it should be the index, so uint64 (probably), since it's included in the block digest

Reply:
I thought this was pk: is the SR25519 session public key associated with the block producer?